sdk/core/core-http/lib/url.ts

```diff
@@ -152,6 +152,8 @@ export class URLQuery {
  */
 export class URLBuilder {
   private _scheme: string | undefined;
+  private _username: string | undefined;
+  private _password: string | undefined;
```

Comment:
AFAIK username and password are not common occurrences in a url. If that is right, and this requirement is solely for proxy urls, then it shouldn't live in `URLBuilder`. Alternative can be to have a `ProxyURLBuilder` that extends `URLBuilder`

Reply:
It is true that currently proxy policy is the only consumer.  `username:password@host` is valid URL format though according to the specification.  If we don't want to add parsing of them to `URLBuilder`, I originally had an idea of a more direct fix in proxy policy: extract and remove user/pass from the url via string manipulation then pass the modified url to `URLBuilder.parse`

Reply:
Submitted PR #7211 for the direct approach.